sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala

```diff
@@ -794,4 +795,28 @@ class CachedTableSuite extends QueryTest with SQLTestUtils with SharedSQLContext
       }
     }
   }
+
+  private def checkIfNoJobTriggered(f: => DataFrame): DataFrame = {
+    var numJobTrigered = 0
+    val jobListener = new SparkListener {
+      override def onJobStart(jobStart: SparkListenerJobStart): Unit = {
+        numJobTrigered += 1
+      }
+    }
+    sparkContext.addSparkListener(jobListener)
+    try {
+      val df = f
+      assert(numJobTrigered === 0)
```

Comment:
before this assert, we should make sure the event queue is empty, via `sparkContext.listenerBus.waitUntilEmpty`